sources/packages/web/src/components/common/Reports.vue

```diff
@@ -56,6 +61,19 @@ export default defineComponent({
         props.reportList,
       );
     };
+    const formChanged = async (form: FormIOForm, event: any) => {
+      formData = form;
+      const reportType = formioUtils.getComponentValueByKey(
+        formData,
+        REPORT_TYPE_DROPDOWN_KEY,
+      );
+      if (
+        event.changed?.component.key === REPORT_TYPE_DROPDOWN_KEY &&
+        reportType === InstitutionReports.OfferingDetails
```

Comment:
Can we rely on something more generic? For instance, if the `REPORT_TYPE_DROPDOWN_KEY` component became visible.

Reply:
I believe you meant `PROGRAM_YEAR_DROPDOWN_KEY` became visible?

Reply:
As we talked here are the changes needed to support the idea and also the new method to unblock the search of the component internal to a container.

Report.vue changes.
```ts
const formChanged = async (form: FormIOForm, event: FormIOChangeEvent) => {
  // Populates the program year select component if required.
  if (event.changed.component.key === REPORT_TYPE_DROPDOWN_KEY) {
    const programYearSelect = formioUtils.getFirstComponent(
      form,
      PROGRAM_YEAR_DROPDOWN_KEY,
    );
    if (
      programYearSelect._visible &&
      !programYearSelect.selectOptions.length
    ) {
      // Load program year data if the select is visible and
      // its items are not populated yet.
      await formioDataLoader.loadProgramYear(
        form,
        PROGRAM_YEAR_DROPDOWN_KEY,
      );
    }
  }
};
```

`useFormioUtils.ts` - `recursiveSearch` method adding the new options.
```ts
const recursiveSearch = (
    form: any,
    matchCondition: (component: any) => boolean,
    options?: {
      stopOnFirstMatch: boolean;
    },
  ): any[] => {
    const matchedComponents: any[] = [];
    internalRecursiveSearch(
      form.components,
      matchedComponents,
      matchCondition,
      options,
    );
    return matchedComponents;
  };
```

`useFormioUtils.ts` - `internalRecursiveSearch` method adding the new options.
```ts
const internalRecursiveSearch = (
    components: any[],
    matchedComponents: any[],
    matchCondition: (component: any) => boolean,
    options?: {
      stopOnFirstMatch: boolean;
    },
  ) => {
    const stopOnFirstMatch = options?.stopOnFirstMatch ?? false;
    components.forEach((component: any) => {
      if (stopOnFirstMatch && matchedComponents.length) {
        // If only the first match is needed, and one was found, stop searching.
        return;
      }
      if (component.components) {
        internalRecursiveSearch(
          component.components,
          matchedComponents,
          matchCondition,
          options,
        );
      }
      if (matchCondition(component)) {
        matchedComponents.push(component);
      }
    });
  };
```

`useFormioUtils.ts` - new method `getFirstComponent`.
```ts
 /**
   * Recursively find a component by its key and returns the first one found.
   * @param form form.io form.
   * @param componentKey form.io API key name.
   * @returns first component found.
   */
  const getFirstComponent = (
    form: FormIOForm,
    componentKey: string,
  ): FormIOComponent => {
    const [firstComponentFound] = recursiveSearch(
      form,
      (component) => component.key === componentKey,
      { stopOnFirstMatch: true },
    );
    return firstComponentFound;
  };
```

Interface changes for `FormIOComponent`.
```ts
export interface FormIOComponent {
  id: string;
  key: string;
  tags: string[];
  components: FormIOComponent[];
  selectOptions: unknown[];
  _visible: boolean;
}
```